index.bs

```diff
@@ -6145,6 +6282,358 @@ However, [=authenticators=] that do not utilize [[!FIDO-CTAP]] do not necessaril
+
+1. Store the |aaguid|, |dpk|, |scope|, |fmt|, |attStmt| values indexed to the <code>|credential|.{{Credential/id}}</code> in the [=user account=].
+
+Issue(w3c/webauthn#1711): Storing this information ought to only occur if the overall registration operation is successful, i.e., crucially, the "encompassing signature" verified correctly.
+
+See also [[#sctn-device-publickey-extension-usage]] for further details.
+
+#### Authentication (`get()`) #### {#sctn-device-publickey-extension-verification-get}
+
+If the `devicePubKey` extension was included on a {{CredentialsContainer/get()|navigator.credentials.get()}} call, then the below verification steps are performed in the context of <a href=#authn-ceremony-verify-extension-outputs>this step</a> of [[#sctn-verifying-assertion]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|.
+
+1. Let |attObjForDevicePublicKey| be the value of the `devicePubKey` member of |clientExtensionResults|.
+
+1. Verify that |attObjForDevicePublicKey| is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to extract the contained fields: |aaguid|, |dpk|, |scope|, |nonce|, |fmt|, |attStmt|.
+
+    Note: The latter |attObjForDevicePublicKey| fields are referenced exclusively in the below steps and are not to be confused with other fields with the same names in other portions of [=authenticator data=].
+
+1. Verify that {{AuthenticationExtensionsDevicePublicKeyOutputs/signature}} is a valid signature over the [=assertion signature=] [input](#fig-signature) by the [=device public key=] |dpk|. (The signature algorithm is the same as for the [=user credential=].)
+
+1. If the [=[RP]=]'s [=user account=] mapped to the <code>|credential|.{{Credential/id}}</code> in play (i.e., for the user being authenticated) holds `aaguid`, `dpk`, `scope`, `fmt`, and `attStmt` values corresponding to the extracted |attObjForDevicePublicKey| fields, then perform binary equality checks between the corresponding stored values and the extracted field values. The [=[RP]=] may have more than one set of {`aaguid`, `dpk`, `scope`, `fmt`, `attStmt`} values mapped to the [=user account=] and <code>|credential|.{{Credential/id}}</code> pair and each set must be checked.
+
+    Issue(w3c/webauthn#1711): In the below steps, where storing the extracted |aaguid|, |dpk|, |scope|, |fmt|, |attStmt| values is called for, it ought to only occur if the overall authentication operation is successful, i.e., crucially, the "encompassing signature" verified correctly.
+
+    1. If more than one stored {`aaguid`, `dpk`, `scope`, `fmt`, `attStmt`} value set matched with the extracted |attObjForDevicePublicKey| fields, then some form of error has occurred. It is indeterminate whether this is a known device. Terminate these verification steps.
+
+    1. Otherwise, if the results are:
+
+    <dl class="switch">
+        : successful
+        :: This is likely a known device:
```

Comment:
It's a bit unclear what exactly is meant by "the results are successful". Maybe rearrange to something like this?

```suggestion
    If the above set of binary equality checks resulted in
    
    <dl class="switch">
        : more than one match
        :: Some form of error has occurred. It is indeterminate whether this is a known device. Terminate these verification steps.

        : exactly one match
        :: This is likely a known device: [...]
        
        : zero matches
        :: This is possibly a new [=device public key=] signifying a new device: [...]
```